specification/communication/data-plane/Chat/stable/2023-11-07/communicationserviceschat.json

```diff
@@ -0,0 +1,1791 @@
+        "topicUpdated",
+        "participantAdded",
+        "participantRemoved"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ChatMessageType",
+        "modelAsString": true
+      }
+    },
+    "SendChatMessageRequest": {
+      "description": "Details of the message to send.",
+      "required": [
+        "content"
+      ],
+      "type": "object",
+      "properties": {
+        "content": {
+          "description": "Chat message content.",
+          "type": "string",
+          "example": "Come one guys, lets go for lunch together."
+        },
+        "senderDisplayName": {
+          "description": "The display name of the chat message sender. This property is used to populate sender name for push notifications.",
+          "type": "string",
+          "example": "Bob Admin"
+        },
+        "type": {
+          "$ref": "#/definitions/ChatMessageType"
+        },
+        "metadata": {
+          "description": "Message metadata.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SendChatMessageResult": {
+      "description": "Result of the send message operation.",
+      "required": [
+        "id"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "A server-generated message id.",
+          "type": "string",
+          "example": "123456789"
+        }
+      }
+    },
+    "ChatParticipant": {
+      "description": "A participant of the chat thread.",
+      "required": [
+        "communicationIdentifier"
+      ],
+      "type": "object",
+      "properties": {
+        "communicationIdentifier": {
+          "$ref": "#/definitions/CommunicationIdentifierModel"
+        },
+        "displayName": {
+          "description": "Display name for the chat participant.",
+          "type": "string",
+          "example": "Bob"
+        },
+        "shareHistoryTime": {
+          "format": "date-time",
+          "description": "Time from which the chat history is shared with the participant. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2020-10-30T10:50:50Z"
+        }
+      }
+    },
+    "AttachmentType": {
+      "description": "The type of attachment.",
+      "enum": [
+        "image"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "AttachmentType",
+        "modelAsString": true
+      }
+    },
+    "ChatAttachment": {
+      "description": "An attachment in a chat message.",
+      "required": [
+        "id",
+        "attachmentType"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Id of the attachment",
+          "type": "string",
+          "example": "A27C7079-626F-4D23-8BEE-582F2C376B5A"
+        },
+        "attachmentType": {
+          "$ref": "#/definitions/AttachmentType"
+        },
```

Comment:
Currently this field can only have one value, "image". What happens when a new type is added? Does this become polymorphic? If so we should describe it as polymorphic from the start.

Reply:
AttachmentType is going to be `image` and `file`, but because we didn't receive customer validation yet for file-sharing feature, so we decide to GA only image for now.
How do we describe to indicate AttachmentType will be polymorphic?

Reply:
You would specify that the `attachmentType` property is a `discriminator` and then have separate schemas for each attachment type. See the OpenAPI docs here

https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#composition-and-inheritance-polymorphism

Reply:
Thanks for sharing the resource, after having a look, it's actually not going to be polymorphism, both `image` and `file` attachmentType have exactly the same ChatAttachment properties, the only difference is will have a different attachmentType